backend.go

```diff
@@ -103,6 +108,9 @@ func Backend() *kubeAuthBackend {
 		),
 	}
 
+	// Set a default http client.
+	b.httpClient = cleanhttp.DefaultPooledClient()
+
```

Comment:
Now that we've moved setting the client's `TLSClientConfig` on config write, we should also consider setting it when we're instantiating the client so that it is also set whenever backend restarts.